WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -1,2 +1,25 @@
 module CompetitionsHelper
+  def competition_message_for_user(competition, user)
+    message = ""
+    if registration = competition.registrations.find_by_user_id(user.id)
+      if registration.accepted?
+        message = "You are registered. "
+      else
+        message = "You are currently on the waiting list. "
+      end
+    end
+    if competition.isConfirmed
+      if competition.showAtAll
+        message += "This competition is confirmed and visible."
+      else
+        message += "This competition is confirmed but not visible."
+      end
```

Comment:
In my personal opinion, that last line is too complex to be readable. LIfting out the expressions would help, but I'd probably go with this:

    if competition.isConfirmed
      how = competition.showAtAll ? 'and' : 'but not'
      message += "This competition is confirmed #{how} visible."
    else
       how = competition.showAtAll ? 'but' : 'and not'
       message += "This competition is not confirmed #{how} visible."
      end
    end

For the first one, I would personally DRY it to `message = "You are " + (registration.accepted? ? "registered. " : "currently on the waiting list. ")`

Also, I didn't want to bog down this tiny change with a lot of tinkering, but since you seem interested... The `message += "blah blah "` pattern has problems. It's (1) ugly and (2) requires each message to end with a ' ' to give separation for the possible next message, thus making it error prone. This is a better pattern:

    messages = []
    messages << "You are registered." if registration.accepted?
    messages << "Look behind you!" if something
    # etc...
    return messages.join(' ')